core/src/main/java/org/infinispan/configuration/parsing/ConfigurationBuilderHolder.java

```diff
@@ -75,6 +75,13 @@ public ConfigurationBuilder getDefaultConfigurationBuilder() {
       }
    }
 
+   public void read(ConfigurationBuilderHolder other) {
+      globalConfigurationBuilder.read(other.globalConfigurationBuilder.build());
+      for (Map.Entry<String, ConfigurationBuilder> entry : other.getNamedConfigurationBuilders().entrySet()) {
+         newConfigurationBuilder(entry.getKey()).read(entry.getValue().build());
```

Comment:
shouldn't check if it exists instead of blinding replace it?
```suggestion
         namedConfigurationBuilders.computeIfAbsent(entry.getKey(), s -> new ConfigurationBuilder()).read(entry.getValue().build());
```